lean4-infoview/src/infoview/messages.tsx

```diff
@@ -143,7 +142,7 @@ export function AllMessages({uri: uri0}: { uri: DocumentUri }) {
 }
 
 /** We factor out the body of {@link AllMessages} which lazily fetches its contents only when expanded. */
-function AllMessagesBody({uri, messages}: {uri: DocumentUri, messages: () => Promise<InteractiveDiagnostic[]>}) {
+function AllMessagesBody({uri, messages}: {uri: DocumentUri, messages: () => Promise<InteractiveDiagnostic[] | undefined>}) {
```

Comment:
I don't see how it can be undefined.  If we look at this line `<AllMessagesBody uri={uri} messages={iDiags} />` and backtrack to the initialization of `iDiags` we can see it comes from `usePausableState` which is initialized to `iDiags0` and `iDiags0` has type Promise<InteractiveDiagnostic[]>.  And while `getInteractiveDiagnostics` can return undefined, it if does do that then this little `or` statement at the end of the line turns that into an empty array: ` || [];` so it is no longer undefined.

Reply:
Ok, makes sense. I added this change because vscode suggested it (maybe because of vscode upgrade) I will revert this and wait for the final code after language client upgrade.